drivers/gles3/storage/texture_storage.cpp

```diff
@@ -389,24 +391,68 @@ static inline Error _get_gl_uncompressed_format(Image::Format p_format, GLenum &
 			r_gl_type = GL_UNSIGNED_SHORT_5_6_5;
 		} break;
 		case Image::FORMAT_RF: {
-			r_gl_internal_format = GL_R32F;
-			r_gl_format = GL_RED;
-			r_gl_type = GL_FLOAT;
+			if (config->float_texture_linear_supported) {
+				r_gl_internal_format = GL_R32F;
+				r_gl_format = GL_RED;
+				r_gl_type = GL_FLOAT;
+			} else {
+				ERR_PRINT("R32 float texture not supported, converting to R16.");
```

Comment:
`ERR_PRINT()` might be a bit spammy in projects using lots of CurveTextures, so maybe we should use `WARN_PRINT_ONCE()` instead. The condition can't change at runtime anyway – if float textures aren't supported when the project starts, they won't be supported afterwards in the same session.

Reply:
We can change this in a followup PR if it ends up being too intrusive